code/math/curve.cpp

```diff
@@ -0,0 +1,241 @@
+ * source.
+ *
+*/ 
+
+#include "math/curve.h"
+#include "parse/parselo.h"
+
+SCP_vector<Curve> Curves;
+
+int curve_get_by_name(SCP_string in_name) {
+	for (int i = 0; i < (int)Curves.size(); i++) {
+		if (Curves[i].name == in_name)
+			return i;
+	}
+
+	return -1;
+}
+
+void parse_curve_table(const char* filename) {
+	try
+	{
+		read_file_text(filename, CF_TYPE_TABLES);
+		reset_parse();
+
+		optional_string("#Curves");
+
+		while (optional_string("$Name:")) {
+			SCP_string name;
+			stuff_string(name, F_NAME);
+
+			int index = curve_get_by_name(name);
+
+			if (index < 0) {
+				Curve curv = Curve(name);
+				curv.ParseData();
+				Curves.push_back(curv);
+			} else {
+				Curves[index].ParseData();
+			}
+		}
+	}
+	catch (const parse::ParseException& e)
+	{
+		mprintf(("TABLES: Unable to parse '%s'!  Error message = %s.\n", filename, e.what()));
+	}
+}
+
+void fill_default_curves() {
+	for (int i = 2; i < 6; i++) {
+		for (int rev = 0; rev < 2; rev++) {
+			SCP_string name("EaseIn");
+			switch (i) {
+			case 2: name += "Quad"; break;
+			case 3: name += "Cubic"; break;
+			case 4: name += "Quart"; break;
+			case 5: name += "Quint"; break;
+			}
+			if (rev)
+				name += "Rev";
+
+			Curves.emplace_back(name);
+			Curves.back().keyframes.push_back(curve_keyframe{ vec2d { 0.0f, rev ? 1.0f : 0.0f}, CurveInterpFunction::Polynomial, (float)i, 1.0f });
+			Curves.back().keyframes.push_back(curve_keyframe{ vec2d { 1.0f, rev ? 0.0f : 1.0f}, CurveInterpFunction::Constant, 0.0f, 0.0f });
+
+			name = "EaseOut";
+			switch (i) {
+			case 2: name += "Quad"; break;
+			case 3: name += "Cubic"; break;
+			case 4: name += "Quart"; break;
+			case 5: name += "Quint"; break;
+			}
+			if (rev)
+				name += "Rev";
+
+			Curves.emplace_back(name);
+			Curves.back().keyframes.push_back(curve_keyframe{ vec2d { 0.0f, rev ? 1.0f : 0.0f}, CurveInterpFunction::Polynomial, (float)i, -1.0f });
+			Curves.back().keyframes.push_back(curve_keyframe{ vec2d { 1.0f, rev ? 0.0f : 1.0f}, CurveInterpFunction::Constant, 0.0f, 0.0f });
+
+			name = "EaseInOut";
+			switch (i) {
+			case 2: name += "Quad"; break;
+			case 3: name += "Cubic"; break;
+			case 4: name += "Quart"; break;
+			case 5: name += "Quint"; break;
+			}
+			if (rev)
+				name += "Rev";
+
+			Curves.emplace_back(name);
+			Curves.back().keyframes.push_back(curve_keyframe{ vec2d { 0.0f, rev ? 1.0f : 0.0f}, CurveInterpFunction::Polynomial, (float)i, 1.0f });
+			Curves.back().keyframes.push_back(curve_keyframe{ vec2d { 0.5f, 0.5f}, CurveInterpFunction::Polynomial, (float)i, -1.0f });
+			Curves.back().keyframes.push_back(curve_keyframe{ vec2d { 1.0f, rev ? 0.0f : 1.0f}, CurveInterpFunction::Constant, 0.0f, 0.0f });
+		}
+	}
+}
+
+void curves_init() {
+
+	Curves.clear();
+	fill_default_curves();
+
+	if (cf_exists_full("curves.tbl", CF_TYPE_TABLES))
+		parse_curve_table("curves.tbl");
+
+	parse_modular_table(NOX("*-crv.tbm"), parse_curve_table);
+}
+
+Curve::Curve(SCP_string in_name)
+{
+	name = in_name;
```

Comment:
The parameter should be `const &`, or alternatively the assignment should use `std::move`